packages/core/src/components/EditableTypography/EditableTypography.tsx

```diff
@@ -138,15 +146,56 @@ const EditableTypography: VibeComponent<EditableTypographyProps, HTMLElement> =
       }
     }
 
-    function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    function handleChange(event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
       setInputValue(event.target.value);
+
+      if (multiline) {
+        resizeTextarea();
+      }
     }
 
     const toggleKeyboardEditMode = useKeyboardButtonPressedFunc(toggleEditMode);
 
     function focus() {
       if (inputRef.current) {
         inputRef.current?.focus();
+
+        if (multiline) {
+          resizeTextarea();
+        }
+      }
+    }
+
+    /* Dynamically resizes the textarea to fit its content */
+    function resizeTextarea() {
+      console.log("*************************resize called, why?", multiline, inputRef.current);
+      if (inputRef.current) {
+        // Temporarily set the height to "auto" to accurately measure the scroll height of the content inside the textarea.
+        setInputHeight("auto");
+
+        requestAnimationFrame(() => {
+          const textarea = inputRef.current as HTMLTextAreaElement;
+
+          if (!textarea) {
+            return;
+          }
+
+          const computedStyle = window.getComputedStyle(textarea);
+
+          // Calculate the appropriate height by taking into account the scrollable content inside the textarea,
+          // as well as the styles applied to it, such as padding and border widths.
+          const lineHeight = parseFloat(computedStyle.lineHeight) || 16;
+          const paddingTop = parseFloat(computedStyle.paddingTop) || 0;
+          const paddingBottom = parseFloat(computedStyle.paddingBottom) || 0;
+          const borderTopWidth = parseFloat(computedStyle.borderTopWidth) || 0;
+          const borderBottomWidth = parseFloat(computedStyle.borderBottomWidth) || 0;
+
+          const newHeight = textarea.scrollHeight + borderTopWidth + borderBottomWidth;
+          const minHeight = lineHeight + paddingTop + paddingBottom + borderTopWidth + borderBottomWidth;
```

Comment:
This feels really fragile, the line height can change and it will break the logic it, the calculation relies on spacings and borders which can also change
There are better stable alternative ways to adjust the height :)

Reply:
couldn't find any, but lets discuss